### PR TITLE
Move CloudAsyncTask out of MainActivtity

### PR DESCRIPTION
**Pull request by szholdiyarov**
Hi, I moved cloudAsynkTask feature outside the MainActivity.

I know the issue was assigned to @vishal0071 but I hope this helps. 

The class AsyncFeature responsible for running tasks in asynchronous  manner. 
It creates not-UI looper where everything will run. 
The entry-point method is execute(Runnable)  which will start the thread and will run until the queue is empty.

The class CloudSyncTask is just a copy of AsynkTask cloudAsyncTask except that it delegates the results thru interface.

I'm open for any suggestions/comments/questions.

### CODE REVIEW

File: app/src/play/java/com/amaze/filemanager/activities/MainActivity.java [hunk 1]
```diff
@@ -166,441 +166,460 @@
...
-    private boolean backPressedToExitOnce = false;
-    private Toast toast = null;
-    private ActionBarDrawerToggle mDrawerToggle;
-    private Intent intent;
-    private GoogleApiClient mGoogleApiClient;
-    private View drawerHeaderLayout;
-    private View drawerHeaderView, indicator_layout;
-    private RoundedImageView drawerProfilePic;
-    private ImageLoader mImageLoader;
+/*
```
Comment (EmmanuelMess) [inline app/src/play/java/com/amaze/filemanager/activities/MainActivity.java]: This is duplicated, check line 1, Android Studio collapses the comment automatically.

**Comment by EmmanuelMess** [comment]
Have you tested? I assigned this to Vishal because to test I need the paid plugin he made.

**Comment by szholdiyarov** [comment]
@EmmanuelMess thanks, i will remove duplicates
yes, i have tested this branch

**Comment by szholdiyarov** [comment]
@EmmanuelMess I removed the duplicate of license definition.

**Comment by vishal0071** [comment]
Even though you tested, it doesn't seem to work for me. It gives error that the cloud calls are made from main thread.
Plus I think your ````AsyncFeature```` class is redundant as ````Handler```` itself manages all the post runnables in the thread's message queue, you don't need to explicitly do that yourself. Just using ````AppConfig#runInBackground(CustomSyncCallbacks)```` from ````CloudSyncTask```` should do the job.

**Comment by szholdiyarov** [comment]
@vishal0071 thanks for testing.
I will have a look at this

**Comment by EmmanuelMess** [comment]
No activity in more than a month, I'm closing, feel free to ask me to reopen if you add something.

**Pull request closed without merging**